crates/cli_bin/tests/apply.rs

```diff
@@ -2015,3 +2015,167 @@ fn ignores_file_in_grit_dir() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn language_option_file_pattern_apply() -> Result<()> {
+    // Keep _temp_dir around so that the tempdir is not deleted
+    let (_temp_dir, dir) = get_fixture("simple_python", false)?;
+
+    // from the tempdir as cwd, run init
+    run_init(&dir.as_path())?;
+
+    // from the tempdir as cwd, run marzano apply
+    let mut apply_cmd = get_test_cmd()?;
+    apply_cmd.current_dir(dir.as_path());
+    // The language option in the pattern file should be ignored
+    apply_cmd.arg("apply").arg("--force").arg("pattern.grit").arg("--language").arg("java");
+    let output = apply_cmd.output()?;
+
+    // Assert that the command executed successfully
+    assert!(
+        output.status.success(),
+        "Command didn't finish successfully: {}",
+        String::from_utf8(output.stderr)?
+    );
+
+    // Read back the main.py file
+    let target_file = dir.join("main.py");
+    let content: String = std::fs::read_to_string(target_file)?;
+
+    // assert that it matches snapshot
+    assert_snapshot!(content);
```

Comment:
No snapshot needed here, just assert that it's unchanged.

Reply:
A snapshot requires maintenance, and doesn't express the intent as clearly. Asserting the final contents against the original comments better captures the intent.

Reply:
You're not listening - you should *NOT* use a snapshot here. A snapshot is unnecessary if all you're asserting is that the content hasn't changed!

Reply:
Tests look good now actually.